server/src/middleware/errorHandler.ts

```
@@ -0,0 +1,78 @@
+ * Handle 404 errors for undefined routes
+ */
+export const notFound = (req: Request, res: Response, next: NextFunction) => {
+  const error: AppError = new Error(`Route ${req.originalUrl} not found`);
+  error.statusCode = 404;
+  next(error);
+};
+
+/**
+ * Async error wrapper to catch async errors in route handlers
+ */
+export const asyncHandler = (fn: Function) => {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)